pkg/apis/core/validation/validation.go

```diff
@@ -2586,9 +2586,27 @@ func validateLifecycle(lifecycle *core.Lifecycle, fldPath *field.Path) field.Err
 	if lifecycle.PreStop != nil {
 		allErrs = append(allErrs, validateHandler(lifecycle.PreStop, fldPath.Child("preStop"))...)
 	}
+	if lifecycle.Type != nil {
+		allErrs = append(allErrs, validateLifecycleType(lifecycle.Type, fldPath.Child("type"))...)
+	}
 	return allErrs
 }
 
+var supportedLifecycleTypes = sets.NewString(string(core.LifecycleTypeSidecar), string(core.LifecycleTypeStandard))
+
+func validateLifecycleType(lifecycleType *core.LifecycleType, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+	switch *lifecycleType {
+	case core.LifecycleTypeSidecar, core.LifecycleTypeStandard:
+		break
+	case "":
```

Comment:
these 2 cases can be combined as you only defined "Standard" and "Sidecar" types

Reply:
sorry I don't follow, can you explain what you mean?

Reply:
Yes - this is an optional field, there can't be a field.Required error

Reply:
yeah, removed that case